### ISSUE TITLE
Hosts file error

### ISSUE THREAD

**Issue by dmagicone** [derekhe/msfs2020-map-enhancement#698]
New, first time installing the 8.0.4 version.  When I click start I get an error "MITM test failed. The modification of the hosts file may not be successful." It happens with any maps I select. 

![error](https://github.com/user-attachments/assets/1c7cc3f3-764e-4dd0-b616-4d541c82da41)

**Comment by derekhe** [comment]
Please have a look: https://github.com/derekhe/msfs2020-map-enhancement/wiki/Manual-EN#85-hosts-file-issues

**Comment by dmagicone** [comment]
It seems to have added some virtualearth.net entries. Its just several lines of space where it put the entries.

**Comment by dmagicone** [comment]
What should it look like?

**Comment by dmagicone** [comment]
Yes it is successful writing to the host file. It adds those line and on exit removes those lines.

**Comment by fsms2020** [comment]
Hello ! I've the same error. I had a licence 1 year ago and since this last release nothing work ! I checked my Antivirus there's no problem. The read only checkbox is disabled on hosts file. I uninstalled the 7.6.2 (this one is the last one used and worked perfectly). I've installed with admin rights the MSFS Map Enhancement Installer 8.0.3.0 then MSFS Map Enhancement Installer 8.0.4.0 (alternatively only the 8.0.4.0) but same message 

![image](https://github.com/user-attachments/assets/1eb6302e-ced4-4683-9fe5-d4ae3e876863)

I checked everything I'm lost :-( can you help please ? @dmagicone is it ok for you?

**Comment by derekhe** [comment]
Could you try 7 again? I didn’t change the way how hosts works but maybe the check has something wrong. Could you also provide your device id in the license page so that I can check the logs?





------------------ Original ------------------
From: fsms2020 ***@***.***&gt;
Date: Fri,Jul 19,2024 10:20 AM
To: derekhe/msfs2020-map-enhancement ***@***.***&gt;
Cc: hesicong ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [derekhe/msfs2020-map-enhancement] Hosts file error (Issue #698)

**Comment by fsms2020** [comment]
Hi @derekhe TY for the quick reply. my device id "QXVCJBEEKDA6RKP039TQQM0TF5A4W0TVAS38PVWVM5D686A5R26G" (license expired since April 2024)

**Comment by dmagicone** [comment]
> Hi @derekhe TY for the quick reply. my device id "QXVCJBEEKDA6RKP039TQQM0TF5A4W0TVAS38PVWVM5D686A5R26G" (licence expired since April 2024)
> 
> When I choose a bing map for example, an error appear saying "MITM test failed...." and the app write these entries in the hosts file
> 
> ```
> 127.0.0.1 kh.ssl.ak.tiles.virtualearth.net
> 127.0.0.1 t3.ssl.ak.tiles.virtualearth.net
> 127.0.0.1 t2.ssl.ak.tiles.virtualearth.net
> 127.0.0.1 t1.ssl.ak.tiles.virtualearth.net
> 127.0.0.1 t0.ssl.ak.tiles.virtualearth.net
> 127.0.0.1 khstorelive.azureedge.net
> ```

I fixed my error by making a new host file. 

Make a new host file on your desktop. You can copy everything that in your current host file without that virtualearth stuff. That stuff should be removed anyway when you close MSFS Enhancement. Now go to your current host file and rename it to something like host-old. You should get a security request from windows, accept it. Now copy the host file you created on the desktop to the folder of your original host file. You should once again get a windows security request. Accept it and after that you should be able to run MSFS Enhancement and click Start with out that error.  This worked for me.

**Comment by fsms2020** [comment]
Thank you @dmagicone but doesn’t work from my side. « You should get a security request from windows » -> I didn’t get any security request from Windows when doing it so I presume my system is less restrictive. Hope @derekhe could help me with logs …

**Comment by fsms2020** [comment]
Hallelujah it works now ! I switched to the version 7.6.2 first time didn't worked but when relaunch it second time it runs ! 
Just for those who have the problem on win11n try this version instead. If you use Norton don't forget to disable the data protection ! (it will not work if you let it activated). thank you @dmagicone for your help and thank you @derekhe for your work and giving this free. I will test the next updates to switch on v8 with a licence.

**Comment by YuuichiKannami2** [comment]
Im having the same problem for the msf2024
But i already tried creating a new hosts file and everything, and i simple can understand what is going on for it not to work
